### PR TITLE
add quote escape tests for filter expressions

### PR DESCRIPTION
**Pull request by gregsdennis**
Addresses https://github.com/gregsdennis/json-everything/issues/728

### CODE REVIEW

**Review by f3ath — COMMENTED** [review]
Pass on my implementation. Any reason why these tests are negative? Seems like adding a positive match would also check that the escaped chars are not missed/mishandled.

**Comment by gregsdennis** [comment]
> Any reason why these tests are negative?

Nope.  Just forgot to update the data.  Will do in a bit (probably tomorrow).

File: tests/filter.json [hunk 1]
```diff
@@ -932,6 +932,30 @@
...
+      "selector" : "$[?@ == 'quoted\\' literal']",
+      "document" : [{"a": 1}],
+      "result": []
+    },
+    {
+      "name": "string literal, escaped double quote in double quotes",
+      "selector" : "$[?@ == \"quoted\\\" literal\"]",
+      "document" : [{"a": 1}],
+      "result": []
     }
```
Comment (He-Pin) [inline tests/filter.json:959]: How about add tests for named selector too

Comment (gregsdennis) [reply tests/filter.json:959]: Those already exist.

**Comment by f3ath** [comment]
Looks good!

**Pull request merged**